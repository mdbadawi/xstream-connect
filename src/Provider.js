// @flow
import React, { PureComponent } from 'react';
import StoreContext from './StoreContext';

type Props = {
    store: { [key: string]: Function },
    children: any
};

class Provider extends PureComponent<Props> {
    render() {
        return (
            <StoreContext.Provider value={this.props.store}>
                {this.props.children}
            </StoreContext.Provider>
        );
    }
}

export default Provider;
