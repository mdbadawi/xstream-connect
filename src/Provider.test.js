import React from 'react';
import renderer from 'react-test-renderer';
import Provider from './Provider';

test('only child is rendered', () => {
  const component = renderer.create(
    <Provider store={{test: 'test'}}>
      <div>hello</div>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test.skip('store props is mandatory', () => {
  expect(() =>
    renderer.create(<Provider><div>hello</div></Provider>)
  ).toThrow();
});

test('store should be present in child\'s context', () => {
  class Child extends React.Component {
    render(){
      return (
        <div>
          {this.context.store.test}
        </div>
      );
    }
  }
  Child.contextTypes = {
    store: React.PropTypes.object.isRequired
  };
  const component = renderer.create(
    <Provider store={{test: 'toto'}}><Child /></Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
