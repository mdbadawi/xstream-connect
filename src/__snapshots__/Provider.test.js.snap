exports[`test only child is rendered 1`] = `
<div>
  hello
</div>
`;

exports[`test store should be present in child's context 1`] = `
<div>
  toto
</div>
`;
